### PR TITLE
[Tests] Add NgraphTestCase test runner class

### PR DESCRIPTION
**Pull request by postrational**

### CODE REVIEW

File: test/util/test_case.hpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            }
+
+            void run();
+
+        protected:
+            std::shared_ptr<Function> m_function;
+            std::unique_ptr<runtime::Backend> m_backend;
+            std::vector<std::shared_ptr<ngraph::runtime::Tensor>> m_input_tensors;
+            std::vector<std::shared_ptr<ngraph::runtime::Tensor>> m_result_tensors;
+            std::vector<std::shared_ptr<ngraph::op::Constant>> m_expected_outputs;
```
Comment (tsocha) [inline test/util/test_case.hpp]: Why expected outputs are stored in `ngraph` constant instead of `vector`?

File: test/util/test_case.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    auto function_results = m_function->get_results();
+    NGRAPH_CHECK(m_expected_outputs.size() == function_results.size(),
+                 "Expected number of outputs is different from the function's number of results.");
+
+    auto handle = m_backend->compile(m_function);
+    handle->call_with_validate(m_result_tensors, m_input_tensors);
+
+    for (int i = 0; i < m_expected_outputs.size(); ++i)
+    {
+        auto result_tensor = m_result_tensors.at(i);
```
Comment (tomdol) [inline test/util/test_case.cpp]: You can use `const auto&` instead here to avoid new instances of `shared_ptr`.

File: test/util/test_case.hpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+namespace ngraph
+{
+    namespace test
+    {
+        class NgraphTestCase
+        {
+        public:
+            NgraphTestCase(std::shared_ptr<Function> function, const std::string& backend_name)
+                : m_function(std::move(function))
```
Comment (tomdol) [inline test/util/test_case.hpp]: In general moving shared pointers is not a common practice. You can change the first parameter of this constructor to `std::shared_ptr&` and remove the `std::move` instead.

File: test/util/test_case.hpp [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+                m_input_tensors.push_back(tensor);
+
+                ++m_input_index;
+            }
+
+            template <typename T>
+            void add_multiple_inputs(const std::vector<std::vector<T>>& vector_of_values)
+            {
+                for (auto const& value : vector_of_values)
```
Comment (tomdol) [inline test/util/test_case.hpp]: East const -> west const? :)

File: test/util/test_case.hpp [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                             "All function results already have expected outputs.");
+
+                auto function_output_type = results.at(m_output_index)->get_element_type();
+                auto function_output_shape = results.at(m_output_index)->get_shape();
+                auto tensor = m_backend->create_tensor(function_output_type, function_output_shape);
+                m_result_tensors.push_back(tensor);
+
+                auto constant_node = std::make_shared<ngraph::op::Constant>(
+                    function_output_type, function_output_shape, values);
+                m_expected_outputs.push_back(constant_node);
```
Comment (tomdol) [inline test/util/test_case.hpp]: You can use `emplace_back` instead and pass the whole `make_shared` call from above directly as its argument.

**Pull request merged**